### PR TITLE
SOR | Docker compose#23

### PR DESCRIPTION
**Pull request by 4TT1L4**
## Summary

Added a docked-compose file to start the SOR with KUPO backend.

Related to: 
 - #23

## Type of Change

- New feature (non-breaking change which adds functionality)
- Documentation

## Checklist

N / A

## Testing

I have tested the docker-compose file locally.

### CODE REVIEW

**Comment by 4TT1L4** [comment]
TODO
 - Add the configuration repo as submodule.
 - Clean up
 - Add documentation

File: Dockerfile [hunk 1]
```diff
@@ -43,7 +43,6 @@ ENV PKG_CONFIG_PATH="/usr/local/lib/pkgconfig:$PKG_CONFIG_PATH"
 # ==================================[ BUILD ]========================================
 WORKDIR /SOR
 
-RUN cabal update
```
Comment (4TT1L4) [inline Dockerfile:46]: We could remove this, but this is not related to the docker-compose topic, so I am going to remove this change from the PR.

**Comment by 4TT1L4** [comment]
@Haluk-GeniusYield 
I took care of the remaining TODO points. Now the PR is ready to be reviewed.

**Comment by SmaugPool** [comment]
There is a missing backslash in the existing Kupo backed instance command that prevents it to work as is.
Maybe this could be fixed in this PR to avoid an issue+PR just for that:

https://github.com/geniusyield/smart-order-router/blob/230c97969760cb7610c0846688bbe4de5268592c/README.md?plain=1#L190

**Comment by 4TT1L4** [comment]
> Maybe this could be fixed in this PR to avoid an issue+PR just for that:

Nice catch. Thanks for the comment.

I took care of this as well, as you proposed it:
 - 8cec30189e42ede4b4612e32cdbb6e527dcf77a2

**Review by Micrograx — APPROVED** [review]
